pykrige/core.py

```diff
@@ -3,62 +3,68 @@
+__doc__ = """
+PyKrige
+=======
+
+Code by Benjamin S. Murphy and the PyKrige Developers
 bscott.murphy@gmail.com
 
-Dependencies:
-    numpy
-    scipy
-
-Functions:
-    _adjust_for_anisotropy(X, y, center, scaling, angle):
-        Returns X_adj array of adjusted data coordinates. Angles are CCW about
-        specified axes. Scaling is applied in rotated coordinate system.
-    _make_variogram_parameter_list(variogram_model, variogram_model_parameters):
-        Makes a list of variogram model parameters in the expected order if the
-        user has provided the model parameters. If not, returns None, which
-        will ensure that the automatic variogram estimation routine is
-        triggered.
-    _initialize_variogram_model(X, y, variogram_model,
-                                variogram_model_parameters, variogram_function,
-                                nlags, weight, coordinates_type):
-        Returns lags, semivariance, and variogram model parameters.
-        Variogram model parameters are estimated if user did not provide them.
-    _variogram_residuals(params, x, y, variogram_function, weight):
-        Called by _calculate_variogram_model.
-    _calculate_variogram_model(lags, semivariance, variogram_model,
-                               variogram_function, weight):
-        Returns variogram model parameters that minimize the RMSE between the
-        specified variogram function and the actual calculated variogram points.
-    _krige(X, y, coords, variogram_function, variogram_model_parameters,
-           coordinates_type):
-        Function that solves the ordinary kriging system for a single specified
-        point. Returns Z value and sigma squared for the specified coordinates.
-        Used in statistics calculation.
-    _find_statistics(X, y, variogram_funtion, variogram_model_parameters,
-                     coordinates_type):
-        Returns the delta, sigma, and epsilon values for the variogram fit.
-        These arrays are used for statistics calculations.
-    calcQ1(epsilon):
-        Returns the Q1 statistic for the variogram fit (see Kitanidis).
-    calcQ2(epsilon):
-        Returns the Q2 statistic for the variogram fit (see Kitanidis).
-    calc_cR(Q2, sigma):
-        Returns the cR statistic for the variogram fit (see Kitanidis).
-    great_circle_distance(lon1, lat1, lon2, lat2):
-        Returns the great circle distance between two arrays of points given in
-        spherical coordinates. Spherical coordinates are expected in degrees.
-        Angle definition follows standard longitude/latitude definition.
-
-References:
+Summary
+-------
+Methods used by multiple classes.
+
+Routines
+-------
+_adjust_for_anisotropy(X, y, center, scaling, angle):
+    Returns X_adj array of adjusted data coordinates. Angles are CCW about
+    specified axes. Scaling is applied in rotated coordinate system.
+_make_variogram_parameter_list(variogram_model, variogram_model_parameters):
```

Comment:
Do we still want to keep these summaries of routines? Essentially that's the same thing as the [docs API page](file:///home/rth/src/PyKrige/doc/_build/html/api.html) except that it takes more effort to manually keep them up to date. Also, there are private functions, so it's not something that is supposed to appear in the Public API (i.e. summary of available functions) as users should not have to use this directly..

Reply:
I'll move the summaries down to each individual function, they're just intended to keep track of what's what internally anyways